nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-processor-configuration.js

```diff
@@ -333,7 +367,11 @@
         }
 
         // relationships
-        processorConfigDto['autoTerminatedRelationships'] = marshalRelationships();
+        processorConfigDto['autoTerminatedRelationships'] = marshalRelationships('terminate');
+        processorConfigDto['retriedRelationships'] = marshalRelationships('retry');
```

Comment:
If there are no relationships being retried, the inputs defining the retry criteria are hidden. However, below we are always including them in the request payload. So if the user changes the retry criteria and then unselects the retry checkbox for all relationships, the inputs are hidden but the new values are saved.

I believe we have the same scenario for Processors that support Event Driven scheduling strategy. If scheduling strategy was Timer Driven and the user opens the configuration dialog and changes the Run Schedule and switches to Event Driven scheduling strategy the Run Schedule control is hidden but the new value isn't included in the request payload.

Reply:
@mcgilman Thanks for pointing this out. I added this bit of logic to only send retry-associated values in the payload if a relationship is selected for retry: https://github.com/apache/nifi/pull/5666/files#diff-ddf6a79c16963065d09b6e220fade7bcc444a155b0f3a91e6585f097440ad112R376-R380